angular-workspace/projects/example-client-app/src/app/customapp/customapp.component.ts

```diff
@@ -120,6 +121,7 @@ export class CustomAppComponent {
         for (let i = 0; i < numberOfRowsToAdd; i++) {
             tableData.push({
                 id: tableData.length.toString(),
+                parentId: (tableData.length > 4 ? ((tableData.length + i) % 4).toString() : undefined),
```

Comment:
This seems more like the intent:
```suggestion
                parentId: (tableData.length >= 4 ? (tableData.length % 4).toString() : undefined),
```